Bug 1949264 - Add git migration compatibility for ./mach try perf

Andrej1198 opened the pull request:
As we are moving away from hg to git ./mach try perf we will no longer be able to get the try url and will be unable to generate perfcompare links. This patch along with the 2 others on treeherder and mozilla central add a hash to the commit message and allows us to get a perfcompare url that finds the commits via that hash This patch is still a work in progress as to properly test I will need the treeherder one to merge first to test properly as currently I am blocked with cross origin requests being not being allowed

Andrej1198 commented:
No need to review, I added an 'on hold', I need  to properly test the patch and to do that I need to get the treeherder changes merged first and pass all linting checks

Andrej1198 commented on src/components/CompareResults/loader.ts:
```diff
@@ -180,11 +181,23 @@ export async function loader({ request }: { request: Request }) {
-  const baseRevFromUrl = url.searchParams.get('baseRev');
+  let baseRevFromUrl = url.searchParams.get('baseRev');
+  let newRevsFromUrl = url.searchParams.getAll('newRev');
+  const baseHashFromUrl = url.searchParams.get('originalHash');
```
I don't know why I did this, I'm re-writing to fix it here and in the treeherder patch

Carla-Moz commented on src/components/CompareResults/loader.ts:
```diff
@@ -180,11 +181,28 @@ export async function loader({ request }: { request: Request }) {
-  const baseRevFromUrl = url.searchParams.get('baseRev');
+  let baseRevFromUrl = url.searchParams.get('baseRev');
+  let newRevsFromUrl = url.searchParams.getAll('newRev');
+  const baseHashFromUrl = url.searchParams.get('baseHash');
```
Could please add a brief comment on why these changes are necessary?

Carla-Moz commented on src/components/CompareResults/loader.ts:
```diff
@@ -180,11 +181,28 @@ export async function loader({ request }: { request: Request }) {
-  const baseRevFromUrl = url.searchParams.get('baseRev');
+  let baseRevFromUrl = url.searchParams.get('baseRev');
+  let newRevsFromUrl = url.searchParams.getAll('newRev');
+  const baseHashFromUrl = url.searchParams.get('baseHash');
```
Where are `baseHashfromUrl` and `newRevsFromUrl` coming from? Not from the inputs in `Compare with Base`, so I'm confused here. I feel these values will always be null and never reach the if condition below to `fetchRevisionFromHash`.

Carla-Moz replied:
Resolved through zoom meeting.

Carla-Moz reviewed (changes requested):
Hi Andrej, thanks for this PR. I have some questions below to help me better review this patch.

Andrej1198 commented:
Hey @Carla-Moz I added some comments in my code and tests to better explain things
I also added a better commit message to properly explain context, and why I am doing this
Finally I also reverted a change we spoke about and re-defined `baseRevFromUrl` and `newRevsFromUrl` back to constants
Let me know if you have any issues with my work :)

Carla-Moz commented on src/components/CompareResults/loader.ts:
```diff
@@ -180,22 +180,32 @@ export async function loader({ request }: { request: Request }) {
+  // with that hash and update the baseRev and newRev
+  let baseRevsFromHash = baseRevFromUrl;
+  let newRevsFromHash = newRevsFromUrl;
+  let usingRevisionsOrHash = true;
```
Nit: Wondering if the naming here could be clearer. Something like  `notUsingHashToFindRevisions` or something of the sort.

Andrej1198 replied:
I've change to use:  `notUsingHashToFindRevisions`

Carla-Moz reviewed (approved):
@Andrej1198 Thanks for the updates. I just have one nit but LGTM!

julienw commented:
I'm coming a bit late: have you considered using a different endpoint? That would make it possible to use a different loader instead of overloading the existing one (while still using the exacty same react components to render the results).

But without trying it I don't know if this is more or less complex.
